src/App.tsx

```diff
@@ -1,26 +1,357 @@
+      userId: 690,
+      title: title.trim(),
+      completed: false,
+    };
+
+    // Add the temporary Todo item to the state immediately
+    setTodoData((prevData: Todo[]): Todo[] => {
+      return [...prevData, tempTodo];
+    });
+
+    setArrayIdsOfLoading(prev => [...prev, tempTodo.id]);
+
+    addTodo({
+      userId: 690,
+      title: title.trim(),
+      completed: false,
+    })
+      .then((data: Todo) => {
+        // Replace the temporary Todo with the actual data from the response
+        setTodoData((prevData: Todo[]): Todo[] => {
+          return prevData.map(todo => (todo.id === tempTodo.id ? data : todo));
+        });
+        setArrayIdsOfLoading(prev => [...prev, data.id]);
+        wait(WAIT_TIME)
+          .then(() => {
+            setTodoInput('');
+            setIsInputDisable(false);
+          })
+          .finally(() => {
+            setArrayIdsOfLoading([]);
+          });
+      })
+      .catch(() => {
+        setShowErrorBox(true);
+        setTextOfError(ERROR_MESSAGE.addError);
+        setIsInputDisable(false);
+
+        // Optionally, remove the temporary Todo item if there's an error
+        setTodoData((prevData: Todo[]): Todo[] => {
+          return prevData.filter(todo => todo.id !== tempTodo.id);
+        });
+      })
+      .finally(() => {});
+  }
+
+  function deletePost(todoId: number) {
+    setArrayIdsOfLoading(prev => [...prev, todoId]);
+
+    deleteTodo(todoId)
+      .then(() => {
+        setTodoData(prevData => prevData.filter(el => el.id !== todoId));
+      })
+      .catch(() => {
+        setShowErrorBox(true);
+        setTextOfError(ERROR_MESSAGE.deleteError);
+      })
+      .finally(() => {
+        setArrayIdsOfLoading([]);
+
+        // Refocus the input field after deletion
+        if (autoFocusRef.current) {
+          autoFocusRef.current.focus();
+        }
+      });
+  }
+
+  function updatePost(todoId: number, info: Todo) {
+    setArrayIdsOfLoading(prevId => [...prevId, todoId]);
+    wait(WAIT_TIME).then(() => {
+      updateTodo(todoId, info)
+        .then((data: Todo[]) => {
+          setTodoData((prevData: Todo[]): Todo[] => {
+            const updatedData = prevData.map(el => {
+              if (el.id === todoId) {
+                return data;
+              }
+
+              return el;
+            });
+
+            return updatedData as Todo[];
+          });
+        })
+        .catch(() => {
+          setShowErrorBox(true);
+          setTextOfError(ERROR_MESSAGE.updateError);
+        })
+        .finally(() => {
+          setArrayIdsOfLoading([]);
+        });
+    });
   }
 
+  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    if (todoInput.trim().length > 0) {
+      addPost(todoInput);
+    } else {
+      setShowErrorBox(true);
+      setTextOfError(ERROR_MESSAGE.emptyError);
+    }
+  }
+
+  function clearAllCopmpleted(e: React.MouseEvent<HTMLButtonElement>) {
```

Comment:
It is important to follow the same way of creating functions everywhere
```suggestion
  const clearAllCopmpleted = (e: React.MouseEvent<HTMLButtonElement>) => {
```